src/node_options-inl.h

```diff
@@ -307,6 +307,13 @@ void OptionsParser<Options>::Parse(
     if (equals_index != std::string::npos)
       original_name += '=';
 
+    {
```

Comment:
IMO should be:
```
for(auto& i: name) {
  if (i == '_') i = '-';
}
```
So there's no need for the scope, and no use of `while` when there's an iterator.
[ES.71](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-for-range)

Reply:
We still have as a rule here to avoid using non-const references., and the rule you linked doesn’t seem to apply here (this is not a for loop to begin with). If you want, this can be

```c++
for (std::string::size_type i = 0; i < name.size(); ++i) {
  if (name[i] == '_') 
    name[i] = '-';
}
```

Reply:
I do like the `for` variant better. It's simple, and probably has comparable performance.

Reply:
P.S. There [ES.72](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-for-while) - Prefer a for-statement to a while-statement when there is an obvious loop variable

Reply:
Which `for` variant? The no-non-const-references rule has a good reason, which is that it should be obvious what is being modified (and in your code that is `name` and not just `i` itself).

Reply:
The one you suggested.

Reply:
> The no-non-const-references rule has a good reason

I have no doubt rules have good reasons. Personally I'm assuming anything that is not `const` is going to be changed, but I'd be happy to discuss this in the context of updating the c++ style guide.

Reply:
> The no-non-const-references rule

Where can I find a reference to that?